### PR TITLE
test: unit test formatNumber

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves
Unit tests for number formatting -> https://github.com/5afe/safe/wiki/How-to-format-amounts

_Note:_ I've used underscore as numeric separator for readability.

## How to test it
Run `yarn test formatNumber`

### CODE REVIEW

File: src/utils/__tests__/formatNumber.test.ts [hunk 1]
```diff
@@ -1,13 +1,149 @@
-describe('Appease Jest', () => {
-  it('adds at least one test to the test file', () => {
-    expect(true).toBe(true)
+import { formatAmount } from '@/utils/formatNumber'
```
Comment (iamacook) [inline src/utils/__tests__/formatNumber.test.ts:1]: Great tests! We should add another that checks that the "Thousands and decimal separators are used according to user's locale."

File: src/utils/formatNumber.ts [hunk 2]
```diff
@@ -57,31 +58,29 @@ const getCurrencyFormatterOptions = (float: number, currency: string): Intl.Numb
   }
 }
 
-// TODO: Add signs in reference to final point of number formatting
-const formatNumber = (float: number, options: Intl.NumberFormatOptions): string => {
-  const VISIBLE_UPPER_LIMIT = 999 * 10 ** 12 // 999T
-
+const formatNumber = (float: number, options: Intl.NumberFormatOptions, keepSign?: boolean): string => {
```
Comment (iamacook) [inline src/utils/formatNumber.ts:61]: We can determine whether the "Eucledian distance of the amount is smaller than 0.00001" without the `keepSign` argument. I don't think it's necessary as we format numbers the same app wide.

File: src/utils/__tests__/formatNumber.test.ts (hunk 1 above)
Comment (DiogoSoaress) [reply src/utils/__tests__/formatNumber.test.ts:1]: We've decided before to go with literal assertions. Please refer to https://github.com/safe-global/safe-react/pull/3920/commits/09a66b717f763b821719e34edcb7a1ceb33b7c9b#r893219929

File: src/utils/formatNumber.ts (hunk 2 above)
Comment (DiogoSoaress) [reply src/utils/formatNumber.ts:61]: I removed the argument and preferred to check if the string starts with the plus sign

**Review by iamacook — APPROVED** [review]

**Pull request merged**